### PR TITLE
[PCBB] Add testcases to verify separated `DSCP_TO_TC_MAP` on leaf router

### PR DESCRIPTION
**Pull request by bingwang-ms**
### Description of PR

Summary:
This PR is to add two new test cases `testQosSaiSeparatedDscpQueueMapping` and `testQosSaiSeparatedDscpToPgMapping` to verify separated `DSCP_TO_TC_MAP` for uplink and downlink on T1.

Code change in sonic-buildimage https://github.com/sonic-net/sonic-buildimage/pull/11569 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
This PR is to add two new test cases `testQosSaiSeparatedDscpQueueMapping` and `testQosSaiSeparatedDscpToPgMapping` to verify separated `DSCP_TO_TC_MAP` for uplink and downlink on T1.

#### How did you do it?
1. Swap syncd docker to syncd-rpc
2. Generate traffic with various DSCP value, and send the packet to DUT from uplink ports or downlink ports
3. Check the increment in each PG and Queue with SAI rpc call.
 
#### How did you verify/test it?
Verified on a TH2 device
```
qos/test_qos_sai.py::TestQosSai::testQosSaiSeparatedDscpToPgMapping[str-7260cx3-acs-7-None-downstream] PASSED                                                                                                                                            [ 50%]
qos/test_qos_sai.py::TestQosSai::testQosSaiSeparatedDscpToPgMapping[str-7260cx3-acs-7-None-upstream]  PASSED      

qos/test_qos_sai.py::TestQosSai::testQosSaiSeparatedDscpQueueMapping[str-7260cx3-acs-7-None-downstream] PASSED                                                                                                                                           [ 50%]
qos/test_qos_sai.py::TestQosSai::testQosSaiSeparatedDscpQueueMapping[str-7260cx3-acs-7-None-upstream] PASSED                                                                                                                                             [100%]
```
#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
T1 specific test cases.

### CODE REVIEW

**Comment by bingwang-ms** [comment]
@stephenxs Can you please help review this PR?

**Review by stephenxs — COMMENTED** [review]
We plan to have different DSCP_TO_TC mapping on downlink/uplink ports in t0 as well. so hardcoding `AZURE_DOWNLINK` is probably not good.
The rest LGTM.

File: tests/common/fixtures/duthost_utils.py [hunk 1]
```diff
@@ -419,3 +420,65 @@ def utils_create_test_vlans(duthost, cfg_facts, vlan_ports_list, vlan_intfs_dict
             ))
     logger.info("Commands: {}".format(cmds))
     duthost.shell_cmds(cmds=cmds)
+
+
+def separated_dscp_to_tc_map_on_uplink(duthost):
+    """
```
[the remark this answers is not in the document]
Comment (bingwang-ms) [reply tests/common/fixtures/duthost_utils.py:427]: Updated. Now the code counts the `dscp_to_tc_map` for each port.

File: tests/common/fixtures/duthost_utils.py [hunk 2]
```diff
@@ -419,3 +420,65 @@ def utils_create_test_vlans(duthost, cfg_facts, vlan_ports_list, vlan_intfs_dict
...
+    Helper function to calculate DSCP to PG map for a port.
+    The map is derived from DSCP_TO_TC_MAP + TC_TO_PG_MAP
+    return a dict like {0:0, 1:1...}
+    """
+    try:
+        port_qos_map = json.loads(duthost.shell("sonic-cfggen -d --var-json 'PORT_QOS_MAP'")['stdout'])
+        dscp_to_tc_map_name = port_qos_map[port]['dscp_to_tc_map'].split('|')[-1].strip(']')
+        tc_to_pg_map_name = port_qos_map[port]['tc_to_pg_map'].split('|')[-1].strip(']')
+        # Load dscp_to_tc_map
+        dscp_to_tc_map = json.loads(duthost.shell("sonic-cfggen -d --var-json 'DSCP_TO_TC_MAP'")['stdout'])[dscp_to_tc_map_name]
```
[the remark this answers is not in the document]
Comment (bingwang-ms) [reply tests/common/fixtures/duthost_utils.py]: Updated. I cache the map with a module level fixture, thanks for the suggestions!

File: tests/qos/qos_sai_base.py [hunk 3]
```diff
@@ -555,6 +563,18 @@ def dutConfig(
                     portIndex = mgFacts["minigraph_ptf_indices"][portName]
                     portIpMap = {'peer_addr': addr["peer_ipv4"]}
                     dutPortIps.update({portIndex: portIpMap})
+                # If the leaf router is using separated DSCP_TO_TC_MAP on uplink/downlink ports.
+                # we also need to test them separately
+                if use_separated_upkink_dscp_tc_map:
+                    neighName = mgFacts["minigraph_neighbors"].get(portName, {}).get("name", "").lower()
```
[the remark this answers is not in the document]
Comment (bingwang-ms) [reply tests/qos/qos_sai_base.py:569]: For now, yes, the PR doesn't include separated uplink/downlink maps test on T0. Once we have separated maps on T0, the similar change is required for T0.

**Review by stephenxs — APPROVED** [review]
LGTM

**Review by lolyu — APPROVED** [review]

**Comment by wangxin** [comment]
@bingwang-ms There are conflicts cherry-picking this PR to 202012 branch. Can you help create a separate PR to the 202012 branch?

**Comment by bingwang-ms** [comment]
Open another PR https://github.com/sonic-net/sonic-mgmt/pull/6817 to backport the change into `202012` branch.

**Pull request merged**